tests/test_transforms/test_math/test_limit_transform.py

```diff
@@ -0,0 +1,155 @@
+    df1["segment"] = ["segment_1"] * periods
+    df1["target"] = np.random.uniform(10, 20, size=periods)
+    df1["non_target"] = np.random.uniform(10, 20, size=periods)
+    df1.loc[5, "target"] = np.NaN
+
+    df2 = pd.DataFrame({"timestamp": pd.date_range("2020-01-01", periods=periods)})
+    df2["segment"] = ["segment_2"] * periods
+    df2["target"] = np.random.uniform(-10, 0, size=periods)
+    df2["non_target"] = np.random.uniform(-10, 0, size=periods)
+    df2.loc[10, "target"] = np.NaN
+
+    df = pd.concat((df1, df2))
+    df = TSDataset.to_dataset(df)
+    tsds = TSDataset(df, freq="D")
+    return tsds
+
+
+@pytest.mark.parametrize(
+    "lower_bound,upper_bound,in_column,transform",
+    [
+        (None, None, "target", transform_no_bounds),
+        (None, 20, "target", transform_right_bound),
+        (-10, None, "target", transform_left_bound),
+        (-10, 20, "target", transform_both_bounds),
+        (None, None, "non_target", transform_no_bounds),
+        (None, 20, "non_target", transform_right_bound),
+        (-10, None, "non_target", transform_left_bound),
+        (-10, 20, "non_target", transform_both_bounds),
+    ],
+)
+def test_fit_transform(
+    ts_check_limit_transform: TSDataset, lower_bound: float, upper_bound: float, in_column: str, transform
+):
+    """Check forward scaled logit transform works correctly on target column"""
+    df_copy = ts_check_limit_transform.to_pandas()
+    df_copy.loc[:, pd.IndexSlice[["segment_1"], in_column]] = df_copy["segment_1"][in_column].apply(
+        transform, lower_bound=lower_bound, upper_bound=upper_bound
+    )
+    df_copy.loc[:, pd.IndexSlice[["segment_2"], in_column]] = df_copy["segment_2"][in_column].apply(
+        transform, lower_bound=lower_bound, upper_bound=upper_bound
+    )
+    preprocess = LimitTransform(in_column=in_column, lower_bound=lower_bound, upper_bound=upper_bound)
+    result = preprocess.fit_transform(ts=ts_check_limit_transform).to_pandas()
+
+    pd.testing.assert_frame_equal(df_copy, result)
+
+
+@pytest.mark.parametrize(
+    "lower_bound,upper_bound,in_column",
+    [
+        (None, None, "target"),
+        (None, 20, "target"),
+        (-10, None, "target"),
+        (-10, 20, "target"),
+        (None, None, "non_target"),
+        (None, 20, "non_target"),
+        (-10, None, "non_target"),
+        (-10, 20, "non_target"),
+    ],
+)
+def test_inverse_transform(ts_check_limit_transform: TSDataset, lower_bound: float, upper_bound: float, in_column: str):
+    """Check that inverse_transform rolls back transform result"""
+    df_copy = ts_check_limit_transform.to_pandas()
+    preprocess = LimitTransform(in_column=in_column, lower_bound=lower_bound, upper_bound=upper_bound)
+    preprocess.fit_transform(ts=ts_check_limit_transform)
+    preprocess.inverse_transform(ts=ts_check_limit_transform)
+    result = ts_check_limit_transform.to_pandas()
+    pd.testing.assert_frame_equal(df_copy, result)
```

Comment:
Why this test passes if we have problems with precision?

Reply:
Because default precision in pd.testing is lower than in data